client/ayon_nuke/api/plugin.py

```diff
@@ -1225,3 +1245,38 @@ def exposed_write_knobs(settings, plugin_name, instance_node):
         instance_node.addKnob(nuke.Text_Knob('', 'Write Knobs'))
     write_node = nuke.allNodes(group=instance_node, filter="Write")[0]
     link_knobs(exposed_knobs, write_node, instance_node)
+
+
+def update_write_node_filepath(created_inst, changes):
+    """Update instance node on context changes.
+    
+    Whenever any of productName, folderPath, task or productType
+    changes then update:
+    - output filepath of the write node
+    - instance node's name to the product name
+    """
+    keys = ("productName", "folderPath", "task", "productType")
+    if not any(key in changes.changed_keys for key in keys):
+        # No relevant changes, no need to update
+        return
+    data = created_inst.data_to_store()
+    # Update values with new formatted path
+    instance_node = created_inst.transient_data["node"]
+    formatting_data = copy.deepcopy(data)
+    write_node = nuke.allNodes(group=instance_node, filter="Write")[0]
+    formatting_data.update({
+        "fpath_template": (
+        "{work}/renders/nuke/{subset}/{subset}.{frame}.{ext}"),
+        "ext": write_node["file_type"].value()
+    })
+    anatomy_filled = format_anatomy(formatting_data)
+
+    # build file path to workfiles
+    fdir = str(
+        anatomy_filled["work"]["default"]["directory"]
+    ).replace("\\", "/")
+    formatting_data["work"] = fdir
+    fpath = StringTemplate(formatting_data["fpath_template"]).format_strict(
+        formatting_data)
+    write_node["file"].setValue(fpath)
+    instance_node["name"].setValue(formatting_data["productName"])
```

Comment:
Please confirm this "name" update occurs when you want it to @moonyuet - currently it will ALSO happen now when changing the folderPath, task or producType whereas previously it only did on change of product name?

Reply:
This is more dependent on the task and productType, guess it would be better to leave it as the previous did.